### PR TITLE
Feature/abd header

### LINKED ISSUE #18: Create UI for header component
### Feature Categories:

- React 

**Describe the feature/Bug your gonna work on**
- Create UI for header component 

**Additional context**
Som_Timer

### LINKED ISSUE #19: Create functionality for header component
### Feature Categories:

- React

**Describe the feature/Bug your gonna work on**

- Add functionality to buttons

### LINKED ISSUE #20: Create tests for header component
### Feature Categories:

- Testing 

**Describe the feature/Bug your gonna work on**
- Add tests for header component
- Try to use TDD to concurrently test and build 

**Additional context**
Add any other context or screenshots about the feature.

### PR DESCRIPTION
**Pull request by Abdeboskey**
# Som Timer Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
* Create `Header` component that will persist across entire application
* Create unit tests for `Header`
* Create initial styling for `Header` for desktop, tablet, and mobile screen sizes.
* Wrap `App` in `Router` in `index.js`
* Install dependencies for react-router-dom

Closes #18, closes #19, closes #20 
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:

## Has This Been Tested?
- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
* The unit tests for this feature are in `src/Components/Header/Header.test.js`
* Run `npm test` to view the test file in action.

# Message for reviewer:
* The nav icons are currently all the same image, the FE team is gonna collaborate on some R&D to find icons that are interesting but also extremely intuitive for the user. 
* Eventually, the icons will represent `Timer (home)`, `Stats`, `Settings`, and `About`

# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas

Screen Shot 2020-10-24 at 8 06 06 PM

### CODE REVIEW

File: src/Components/Header/Header.module.scss [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+.icon {
+  height: 60px; 
+  filter: drop-shadow(0px 0px 5px $darkBlue);
+
+  &:hover {
+    cursor: pointer;
+  }
+}
+
+@media screen and (max-width: 992px) {
```
Comment (rwilliams659) [inline src/Components/Header/Header.module.scss:49]: Really like how you did the media queries

File: src/Components/Header/Header.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      </MemoryRouter>
+    )
+
+    const title = getByRole('heading', { name: /som timer/i })
+    const settingsButton = getByRole('button', {name: /settings/i})
+    const statsButton = getByRole('button', {name: /stats/i})
+    const timerButton = getByRole('button', {name: /timer/i})
+    const aboutButton = getByRole('button', {name: /about/i})
+    
+    expect(title).toBeInTheDocument()
```
Comment (rwilliams659) [inline src/Components/Header/Header.test.js:21]: Thanks for being willing to compromise & move these into 1 test!

File: src/Components/Header/Header.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    <header className={style.header}>
+      <Link className={style.title} to="/">
+        <h1 className={style.text}>Som Timer</h1>
+      </Link>
+      <nav className={style.navBar}>
+        <NavLink to="/" activeClassName={style.selected}>
+          <button className={style.navButton} aria-label="timer">
+            <img
+              className={style.icon}
+              src={metatronsSettings}
```
Comment (rwilliams659) [inline src/Components/Header/Header.js:17]: Excited to see what these icons look like in the end!

**Review by rwilliams659 — APPROVED** [review]
Nice job, excited to have the header incorporated into the codebase so this starts to look like an actual webapp!

File: src/Components/Header/Header.module.scss (hunk 1 above)
Comment (Abdeboskey) [reply src/Components/Header/Header.module.scss:49]: Thank you for finding these breakpoints!

Comment (jkwest-93) [reply src/Components/Header/Header.module.scss:51]: I agree- the header looks so great in all these different views. Well designed!

**Review by jkwest-93 — APPROVED** [review]
Great work, Aaron!! This looks awesome and really sets us up to hook up these various pieces.

**Review by sciencefixion — APPROVED** [review]
Looks great as always! I can't wait to see all this in action!

**Pull request merged**